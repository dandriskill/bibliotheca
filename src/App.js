// * Developer's Note: This component utilizes the new React Hooks API rather than class-based components
import React, { useState, useEffect } from 'react';
import './styles/App.css';

import Subheader from './components/Subheader';
import BooksHeader from './components/BooksHeader';
import Library from './components/Library';
import StandardModal from './components/StandardModal';
import Success from './components/Success';
import {
  getBooks,
  saveBook,
  updateBook,
  deleteBook,
} from './api';
import { testBooks } from './test-data';

// TODO:
// Write tests
// Style book update/save forms
// Connect to Firebase and write Firebase API calls

// MAYBE:
// Format search bar
// Move props and handlers through Redux?
// Add search bar functionality, which updates the books array by comparing string input to author or title
function App() {
  const [books, setBooks] = useState([]);
  const [booksFallback, setBooksFallback] = useState([]); // Note: Maintains book list after search bar is cleared
  const [modalIsOpen, setModalIsOpen] = useState(false);
  const [modalContent, setModalContent] = useState(null);

  useEffect(() => {
    handleGetBooks();
  }, []);

  function handleGetBooks() {
    setBooks(testBooks);
    setBooksFallback(testBooks);
  }

  function handleSaveBook(book) {
    saveBook(book);
    handleCloseModal();
    handleOpenModal(
      <Success
        message={'This volume has been added!'}
        closeModal={handleCloseModal}
      />
    );
  }

  function handleUpdateBook(book) {
    updateBook(book);
    handleCloseModal();
    handleOpenModal(
      <Success
        message={'This volume has been updated!'}
        closeModal={handleCloseModal}
      />
    );
  }

  function handleDeleteBook(id) {
    deleteBook(id);
    handleCloseModal();
    handleOpenModal(
      <Success
        message={'This volume has been deleted.'}
        closeModal={handleCloseModal}
      />
    );
  }

  function handleOpenModal(content) {
    setModalContent(content);
    setModalIsOpen(true);
  }

  function handleCloseModal() {
    setModalIsOpen(false);
    setModalContent(null);
  }

  return (
    <div className='App'>
      <div className='App-inner'>
        <h1>BIBLIOTHECA</h1>
        <h3>The Library App</h3>
        <div className='divider-container'>
          <div className='divider'></div>
        </div>
        <h1 className='faded-text'>Î©</h1>
        <Subheader
          books={books}
          openModal={handleOpenModal}
          closeModal={handleCloseModal}
          saveBook={handleSaveBook}
        />
        <BooksHeader />
        <Library
          books={books} 
          openModal={handleOpenModal}
          closeModal={handleCloseModal}
          updateBook={handleUpdateBook}
          deleteBook={handleDeleteBook}
        />
      </div>
      <StandardModal
        modalIsOpen={modalIsOpen}
        closeModal={handleCloseModal}
        content={modalContent}
      />
    </div>
  );
}

export default App;
