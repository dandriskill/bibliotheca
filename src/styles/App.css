.App {
  display: flex;
  justify-content: center;
  background-image : url(http://api.thumbr.it/whitenoise-361x370.png?background=grey&noise=5c5c5c&density=5&opacity=10);
  height: 100vh;
  overflow: hidden;
}
.App-inner {
  text-align: center;
  background-image : url(http://api.thumbr.it/whitenoise-361x370.png?background=fffdf7&noise=5c5c5c&density=5&opacity=10);
  color: #282c34;
  height: 100vh;
  width: 80%;
  border-right: 1px solid #1c1f24;
  border-left: 1px solid #1c1f24;
}

.faded-text {
  color: #ebe9e4;
}

.header {
  width: 30%;
  display: flex;
  justify-content: space-around;
}

.divider-container {
  display: flex;
  justify-content: center;
}

.divider {
  width: 30%;
  border-bottom: 2px solid #282c34;
}
