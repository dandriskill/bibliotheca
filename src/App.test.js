import React from 'react';
import { render } from '@testing-library/react';
import App from './App';

describe('App', ()=> {
  test('renders App', () => {
    const { getByText } = render(<App />);
    const appTitleNode = getByText(/bibliotheca/i);
    expect(appTitleNode).toBeInTheDocument();
  });

  test('matches snapshot', () => {
    const container = render(<App />);
    expect(container).toMatchSnapshot();
  });
});
