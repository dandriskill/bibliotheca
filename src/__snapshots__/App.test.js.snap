// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <div
          class="App-inner"
        >
          <h1>
            BIBLIOTHECA
          </h1>
          <h3>
            The Library App
          </h3>
          <div
            class="divider-container"
          >
            <div
              class="divider"
            />
          </div>
          <h1
            class="faded-text"
          >
            Ω
          </h1>
          <div
            class="Subheader"
          >
            <h2
              class="volumes-count"
            >
              Volumes: 
              0
            </h2>
            <span
              class="search-bar"
            >
              <div
                class="SearchBar"
              >
                <input
                  class="searchbar-inner"
                  placeholder="Search by title or author"
                  type="text"
                />
              </div>
            </span>
            <button
              class="add-button"
              type="button"
            >
              Add +
            </button>
          </div>
          <div
            class="BooksHeader"
          >
            <div
              class="books-header-title"
            >
              Title
            </div>
            <div
              class="books-header-author"
            >
              Author
            </div>
            <div
              class="books-header-pages"
            >
              Pages
            </div>
            <div
              class="books-header-available"
            >
              Available
            </div>
            <div
              class="books-header-overdue"
            >
              Overdue
            </div>
            <div
              class="books-header-tools"
            />
          </div>
          <div
            class="Library"
          >
            <div
              class="fallback-text"
            >
              No volumes available.
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ReactModalPortal"
    />
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <div
        class="App-inner"
      >
        <h1>
          BIBLIOTHECA
        </h1>
        <h3>
          The Library App
        </h3>
        <div
          class="divider-container"
        >
          <div
            class="divider"
          />
        </div>
        <h1
          class="faded-text"
        >
          Ω
        </h1>
        <div
          class="Subheader"
        >
          <h2
            class="volumes-count"
          >
            Volumes: 
            0
          </h2>
          <span
            class="search-bar"
          >
            <div
              class="SearchBar"
            >
              <input
                class="searchbar-inner"
                placeholder="Search by title or author"
                type="text"
              />
            </div>
          </span>
          <button
            class="add-button"
            type="button"
          >
            Add +
          </button>
        </div>
        <div
          class="BooksHeader"
        >
          <div
            class="books-header-title"
          >
            Title
          </div>
          <div
            class="books-header-author"
          >
            Author
          </div>
          <div
            class="books-header-pages"
          >
            Pages
          </div>
          <div
            class="books-header-available"
          >
            Available
          </div>
          <div
            class="books-header-overdue"
          >
            Overdue
          </div>
          <div
            class="books-header-tools"
          />
        </div>
        <div
          class="Library"
        >
          <div
            class="fallback-text"
          >
            No volumes available.
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
